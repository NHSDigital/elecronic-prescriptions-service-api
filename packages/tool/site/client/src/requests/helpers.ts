export interface HubResponse {
  certificate: string,
  status_code: number,
  status_string: string,
  message: string,
  timestamp: string,
  signatures: Array<Signature>,
  failed_signatures: Array<Signature>
}

export interface Signature {
  id: string,
  signature: string
}

export function getToken(queryString: URLSearchParams): string {
  const token = queryString.get("token")

  if (!token) {
    throw new TypeError("Token not present in query string.")
  }

  return token
}

export function getState(queryString: URLSearchParams): string | null {
  return queryString.get("state")
}

export function getQueryString(): URLSearchParams {
  const queryString = window.location.search
  if (!queryString) {
    throw new TypeError("Empty query string.")
  }

  return new URLSearchParams(queryString)
}

export async function start(signFn: (jwt: string) => Promise<HubResponse>): Promise<void> {
  try {
    //showStatus("Signing in progress", "Downloading the signature request from the Signing Service.")
    const queryString = getQueryString()
    const token = getToken(queryString)
    const signatureRequest = await retrieveSignatureRequestFromServer(token)

    //showStatus("Signing in progress", "Communicating with Credential Management to generate a signature.")
    const hubResponse = await signFn(signatureRequest)
    validateHubResponse(hubResponse)

    //showStatus("Signing in progress", "Uploading the signature to the Signing Service.")
    const redirectUrl = await sendResponseToProvider(hubResponse, token)

    //showStatus("Signing complete", "Sending you back to the calling application.")
    const state = getState(queryString)
    window.location.href = state ? `${redirectUrl}&state=${state}` : redirectUrl
  } catch (e) {
    console.log(e)
    showError(e.message)
  }
}

function validateHubResponse(hubResponse: HubResponse) {
  if (!hubResponse) {
    throw new TypeError("No response from Credential Management.")
  }

  if (hubResponse.status_code !== 0) {
    console.log(`Got error response from Credential Management: ${JSON.stringify(hubResponse)}`)
    throw new TypeError("Error response from Credential Management.")
  }

  if (!hubResponse.certificate || !hubResponse.signatures?.length) {
    console.log(`Got invalid response from Credential Management: ${JSON.stringify(hubResponse)}`)
    throw new TypeError("Invalid response from Credential Management.")
  }
}

async function retrieveSignatureRequestFromServer(token: string): Promise<string> {
  const response = await fetch(`./client/signaturerequest/${token}`, {
    headers: {
      "X-CSRF-Token": getCSRFTokenFromCookie()
    }
  })
  if (response.status !== 200) {
    console.error(`Got unexpected response from Signing Service: ${response.status}`)
    throw new Error("Failed to retrieve signature request from Signing Service.")
  }
  const responseText = await response.text()
  if (!responseText) {
    throw new Error("Failed to parse response from Signing Service.")
  }
  return responseText
}

function getCSRFTokenFromCookie(): string {
  const cookies = getCookies()
  const crumb = cookies.get("crumb")

  if (!crumb) {
    throw new Error("Could not find CSRF token")
  }

  return crumb
}

function getCookies(): Map<string, string> {
  try {
    return new Map(document.cookie.split(";").map(cookie => {
      const cookieKey = cookie.split("=")[0].trim()
      const cookieValue = cookie.split("=")[1].trim()

      return [cookieKey, cookieValue]
    }))
  } catch(e) {
    console.error("Could not read cookies")
    console.error(e)
    return new Map()
  }
}

// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
const element = (identifier: string) => document.getElementById(identifier)!

export function showError(details?: string): void {
  element("status-container").style.display = "none"
  if (details) {
    element("error-details").innerText = `Error details: ${details}`
  }
  element("error-container").style.display = "block"
}

export async function sendResponseToProvider(hubResponse: HubResponse, token: string): Promise<string> {
  const response = await fetch(
    `./client/signatureresponse/${token}`,
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        "X-CSRF-Token": getCSRFTokenFromCookie()
      },
      body: JSON.stringify({
        "signatures": hubResponse.signatures,
        "certificate": hubResponse.certificate
      })
    }
  )

  if (response.status !== 200) {
    console.error(`Got unexpected response from Signing Service: ${response.status}`)
    throw new Error("Failed to upload signature to Signing Service.")
  }

  const responseJson = await response.json()
  if (!responseJson) {
    throw new Error("Failed to parse response from Signing Service.")
  }

  return responseJson.redirectUri
}
