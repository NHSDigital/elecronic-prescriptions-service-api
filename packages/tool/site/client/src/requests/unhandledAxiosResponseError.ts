import {AxiosResponse} from "axios"

export class UnhandledAxiosResponseError extends Error {
  response: AxiosResponse

  constructor(message: string, response: AxiosResponse) {
    super(message)
    this.response = response
  }
}
