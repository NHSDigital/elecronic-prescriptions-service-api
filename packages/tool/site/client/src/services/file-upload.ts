import {Dispatch, SetStateAction} from "react"

export function readPrescriptionsFromFiles(
  // eslint-disable-next-line no-undef
  files: FileList,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  prescriptionFilesUploaded: Array<any>,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  setPrescriptionFilesUploaded: Dispatch<SetStateAction<Array<any>>>
): void {
  for (let i = 0; i < files.length; i++) {
    // eslint-disable-next-line no-undef
    const reader = new FileReader()
    reader.onload = event => {
      const result = event.target.result
      prescriptionFilesUploaded.push(result)
      setPrescriptionFilesUploaded(prescriptionFilesUploaded)
    }
    reader.readAsText(files[i])
  }
}
