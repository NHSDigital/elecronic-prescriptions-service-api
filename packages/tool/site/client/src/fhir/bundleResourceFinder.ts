import * as fhir from "fhir/r4"
import {MedicationDispense, MedicationRequest} from "./helpers"

export const getPatientResources = buildResourceFinder<fhir.Patient>("Patient")
export const getPractitionerResources = buildResourceFinder<fhir.Practitioner>("Practitioner")
export const getPractitionerRoleResources = buildResourceFinder<fhir.PractitionerRole>("PractitionerRole")
export const getOrganizationResources = buildResourceFinder<fhir.Organization>("Organization")
export const getMedicationRequestResources = buildResourceFinder<MedicationRequest>("MedicationRequest")
export const getMedicationDispenseResources = buildResourceFinder<MedicationDispense>("MedicationDispense")
export const getMessageHeaderResources = buildResourceFinder<fhir.MessageHeader>("MessageHeader")
export const getCommunicationRequestResources = buildResourceFinder<fhir.CommunicationRequest>("CommunicationRequest")

function buildResourceFinder<T extends fhir.FhirResource>(resourceType: string): (payload: fhir.Bundle) => Array<T> {
  return function (bundle: fhir.Bundle): Array<T> {
    const typeCheck = (resource: fhir.FhirResource): resource is T => resource.resourceType === resourceType
    return bundle.entry.map(entry => entry.resource).filter(typeCheck)
  }
}
