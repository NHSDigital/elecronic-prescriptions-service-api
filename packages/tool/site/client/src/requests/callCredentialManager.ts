import {HubResponse, start} from "./helpers"

// eslint-disable-next-line @typescript-eslint/no-explicit-any
(window as any).jQuery = jQuery;
// eslint-disable-next-line @typescript-eslint/no-explicit-any
(window as any).$ = jQuery

const SIGNALR_URL = `"http://localhost:"+prService.portNumber()+"/signalr"`
const SIGNALR_HUB_NAME = "signingHub"
const SIGNALR_METHOD_NAME = "requestToSign"

export async function sign(jwt: string): Promise<HubResponse> {
  const connection = $.hubConnection(SIGNALR_URL, {logging: true})
  if (connection.state !== SignalR.ConnectionState.Disconnected) {
    console.log(`Unexpected SignalR connection state ${connection.state}`)
    throw new Error("SignalR connection is already active.")
  }

  try {
    const proxy = connection.createHubProxy(SIGNALR_HUB_NAME)
    await connection.start({transport: "longPolling"})
    return await proxy.invoke(SIGNALR_METHOD_NAME, jwt)
  } finally {
    connection.stop()
  }
}

window.addEventListener("DOMContentLoaded", () => {
  start(sign)
})
