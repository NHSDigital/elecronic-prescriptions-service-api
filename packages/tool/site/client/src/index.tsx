import * as React from "react"
import {Page} from "./components/page"
import SignPage from "./pages/signPage"
import * as ReactDOM from "react-dom"
import {BrowserRouter, Route, Switch} from "react-router-dom"
import ClaimPage from "./pages/claimPage"
import DispensePage from "./pages/dispensePage"
import {CookiesProvider} from "react-cookie"
import SendPage from "./pages/sendPage"
import HomePage from "./pages/homePage"
import PrescriptionSearchPage from "./pages/prescriptionSearchPage"
import ReleasePage from "./pages/releasePage"
import CancelPage from "./pages/cancelPage"
import MyPrescriptionsPage from "./pages/myPrescriptionsPage"
import LoginPage from "./pages/loginPage"
import ValidatePage from "./pages/validatePage"
import ReturnPage from "./pages/returnPage"
import WithdrawPage from "./pages/withdrawPage"
import LoadPage from "./pages/loadPage"
import {Environment} from "./services/environment"
import LogoutPage from "./pages/logoutPage"
import ConfigPage from "./pages/configPage"
import ComparePage from "./pages/comparePage"
import ViewPrescriptionPage from "./pages/viewPrescriptionPage"
import {PageFullWidth} from "./components/pageFullWidth"
import {PageContainer} from "./components/pageContainer"
import DoseToTextPage from "./pages/doseToTextPage"
import TrackerViewPrescriptionPage from "./pages/trackerViewPrescription"

const customWindow = window as Record<string, any>

export interface AppContextValue {
  baseUrl: string
  environment: Environment
}

export const AppContext = React.createContext<AppContextValue>({baseUrl: "/", environment: "int"})

async function startApplication(baseUrl: string, environment: Environment): Promise<void> {
  const urlParams = new URLSearchParams(window.location.search)
  const content = (
    <AppContext.Provider value={{baseUrl, environment}}>
      <CookiesProvider>
        <Page>
          <BrowserRouter>
            <Switch>
              <Route path={`${baseUrl}/`}>
                <PageContainer>
                  <HomePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dose-to-text`}>
                <PageContainer>
                  <DoseToTextPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}config`}>
                <PageContainer>
                  <ConfigPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}my-prescriptions`}>
                <PageContainer>
                  <MyPrescriptionsPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}login`}>
                <PageContainer>
                  <LoginPage separateAuth={urlParams.get("separate_auth")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}logout`}>
                <PageContainer>
                  <LogoutPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/load`}>
                <PageContainer>
                  <LoadPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/edit`}>
                <PageContainer>
                  <SignPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/send`}>
                <PageContainer>
                  <SendPage token={urlParams.get("token")} state={urlParams.get("state")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/cancel`}>
                <PageContainer>
                  <CancelPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/release`}>
                <PageContainer>
                  <ReleasePage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/return`}>
                <PageContainer>
                  <ReturnPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/withdraw`}>
                <PageContainer>
                  <WithdrawPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/dispense`}>
                <PageContainer>
                  <DispensePage prescriptionId={urlParams.get("prescription_id")} amendId={urlParams.get("amend_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/claim`}>
                <PageContainer>
                  <ClaimPage prescriptionId={urlParams.get("prescription_id")} amend={!!urlParams.get("amend")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}search`}>
                <PageContainer>
                  <PrescriptionSearchPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}tracker`}>
                <PageContainer>
                  <TrackerViewPrescriptionPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}view`}>
                <PageContainer>
                  <ViewPrescriptionPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}validate`}>
                <PageContainer>
                  <ValidatePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}compare-prescriptions`}>
                <PageFullWidth>
                  <ComparePage />
                </PageFullWidth>
              </Route>
            </Switch>
          </BrowserRouter>
        </Page>
      </CookiesProvider>
    </AppContext.Provider>
  )
  ReactDOM.render(content, document.getElementById("root"))
}

customWindow.startApplication = startApplication
