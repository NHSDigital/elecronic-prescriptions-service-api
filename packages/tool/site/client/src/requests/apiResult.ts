import {FhirResource} from "fhir/r4"

export function isApiResult(data: unknown): data is ApiResult {
  const result = data as ApiResult
  return typeof result.success === "boolean"
    && "request" in result
    && "response" in result
}

export interface ApiResult {
  success: boolean
  request: FhirResource
  request_xml?: string
  response: FhirResource
  response_xml?: string
}
