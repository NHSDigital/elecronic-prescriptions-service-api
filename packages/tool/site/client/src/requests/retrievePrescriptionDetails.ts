/* eslint-disable max-len */
import {axiosInstance} from "./axiosInstance"
import * as fhir from "fhir/r4"
import {getResponseDataIfValid} from "./getValidResponse"
import {getArrayTypeGuard, isBundle} from "../fhir/typeGuards"

export async function getPrescriptionOrderMessage(baseUrl: string, prescriptionId: string): Promise<fhir.Bundle> {
  const prescriptionOrderResponse = await axiosInstance.get<fhir.Bundle>(`${baseUrl}dispense/release/${prescriptionId}`)
  return getResponseDataIfValid(prescriptionOrderResponse, isBundle)
}

export async function getDispenseNotificationMessages(baseUrl: string, prescriptionId: string): Promise<Array<fhir.Bundle>> {
  const dispenseNotificationsResponse = await axiosInstance.get<Array<fhir.Bundle>>(`${baseUrl}dispenseNotifications/${prescriptionId}`)
  return getResponseDataIfValid(dispenseNotificationsResponse, getArrayTypeGuard(isBundle))
}
