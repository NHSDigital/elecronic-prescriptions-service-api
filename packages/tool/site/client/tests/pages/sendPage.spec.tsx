import {waitFor} from "@testing-library/react"
import {screen} from "@testing-library/dom"
import pretty from "pretty"
import * as React from "react"
import moxios from "moxios"
import {AppContextValue} from "../../src"
import {renderWithContext} from "../renderWithContext"
import SendPage from "../../src/pages/sendPage"
import {axiosInstance} from "../../src/requests/axiosInstance"
import {internalDev} from "../../src/services/environment"
import {readBundleFromFile} from "../messages"
import {getMedicationRequestResources} from "../../src/fhir/bundleResourceFinder"

const baseUrl = "baseUrl/"
const token = "MzQxMWJmMjUtMDNlMy00N2FiLWEyOGItMGIyYjVlNTg4ZGU3"
const context: AppContextValue = {baseUrl, environment: internalDev}

const prescriptionsUrl = `${baseUrl}prescriptions`
const sendUrl = `${baseUrl}api/prescribe/send`

const prescriptionOrder = readBundleFromFile("prescriptionOrder.json")

beforeEach(() => moxios.install(axiosInstance))

afterEach(() => moxios.uninstall(axiosInstance))

test("Displays confirmation page if single prescription is sent successfully", async () => {
  const prescriptionId = "003D4D-A99968-4C5AAJ"
  const prescription = clone(prescriptionOrder)
  prescription.id = "1"
  getMedicationRequestResources(prescription)[0].groupIdentifier.value = prescriptionId

  moxios.stubRequest(prescriptionsUrl, {
    status: 200,
    response: [prescription]
  })

  moxios.stubRequest(sendUrl, {
    status: 200,
    response: {
      results: [{
        prescription_id: prescriptionId,
        bundle_id: "1",
        request: {},
        request_xml: "XML Request",
        response: {},
        response_xml: "XML Response",
        success: true
      }]
    }
  })

  const container = await renderPage()

  await waitFor(() => screen.getByText(prescriptionId))
  expect(screen.getByText("XML Request")).toBeTruthy()
  expect(screen.getByText("XML Response")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays confirmation page if multiple prescriptions are sent successfully", async () => {
  const prescriptionId1 = "003D4D-A99968-4C5AAJ"
  const prescriptionId2 = "008070-A99968-41CD9V"
  const prescriptionId3 = "010E34-A99968-467D9Z"
  const prescription1 = clone(prescriptionOrder)
  prescription1.id = "1"
  getMedicationRequestResources(prescription1)[0].groupIdentifier.value = prescriptionId1
  const prescription2 = clone(prescriptionOrder)
  prescription2.id = "2"
  getMedicationRequestResources(prescription2)[0].groupIdentifier.value = prescriptionId2
  const prescription3 = clone(prescriptionOrder)
  prescription3.id = "3"
  getMedicationRequestResources(prescription3)[0].groupIdentifier.value = prescriptionId3

  moxios.stubRequest(prescriptionsUrl, {
    status: 200,
    response: [prescription1, prescription2, prescription3]
  })

  moxios.stubRequest(sendUrl, {
    status: 200,
    response: {
      results: [
        {
          prescription_id: prescriptionId1,
          bundle_id: "1",
          success: true
        },
        {
          prescription_id: prescriptionId2,
          bundle_id: "2",
          success: true
        },
        {
          prescription_id: prescriptionId3,
          bundle_id: "3",
          success: true
        }
      ]
    }
  })

  const container = await renderPage()

  expect(screen.getByText(prescriptionId1)).toBeTruthy()
  expect(screen.getByText(prescriptionId2)).toBeTruthy()
  expect(screen.getByText(prescriptionId3)).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Exception report button not shown if there are pending prescriptions", async () => {
  const prescriptionId1 = "003D4D-A99968-4C5AAJ"
  const prescriptionId2 = "008070-A99968-41CD9V"
  const prescriptionId3 = "010E34-A99968-467D9Z"
  const prescription1 = clone(prescriptionOrder)
  prescription1.id = "1"
  getMedicationRequestResources(prescription1)[0].groupIdentifier.value = prescriptionId1
  const prescription2 = clone(prescriptionOrder)
  prescription2.id = "2"
  getMedicationRequestResources(prescription2)[0].groupIdentifier.value = prescriptionId2
  const prescription3 = clone(prescriptionOrder)
  prescription3.id = "3"
  getMedicationRequestResources(prescription3)[0].groupIdentifier.value = prescriptionId3

  moxios.stubRequest(prescriptionsUrl, {
    status: 200,
    response: [prescription1, prescription2, prescription3]
  })

  moxios.stubRequest(sendUrl, {
    status: 200,
    response: {
      results: [
        {
          prescription_id: prescriptionId1,
          bundle_id: "1",
          success: true
        },
        {
          prescription_id: prescriptionId2,
          bundle_id: "2",
          success: "unknown"
        },
        {
          prescription_id: prescriptionId3,
          bundle_id: "3",
          success: false
        }
      ]
    }
  })

  const container = await renderPage()

  expect(screen.getByText(prescriptionId1)).toBeTruthy()
  expect(screen.getByText(prescriptionId2)).toBeTruthy()
  expect(screen.getByText(prescriptionId3)).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Exception report button shown if there are failed prescriptions", async () => {
  const prescriptionId1 = "003D4D-A99968-4C5AAJ"
  const prescriptionId2 = "008070-A99968-41CD9V"
  const prescriptionId3 = "010E34-A99968-467D9Z"
  const prescription1 = clone(prescriptionOrder)
  prescription1.id = "1"
  getMedicationRequestResources(prescription1)[0].groupIdentifier.value = prescriptionId1
  const prescription2 = clone(prescriptionOrder)
  prescription2.id = "2"
  getMedicationRequestResources(prescription2)[0].groupIdentifier.value = prescriptionId2
  const prescription3 = clone(prescriptionOrder)
  prescription3.id = "3"
  getMedicationRequestResources(prescription3)[0].groupIdentifier.value = prescriptionId3

  moxios.stubRequest(prescriptionsUrl, {
    status: 200,
    response: [prescription1, prescription2, prescription3]
  })

  moxios.stubRequest(sendUrl, {
    status: 200,
    response: {
      results: [
        {
          prescription_id: prescriptionId1,
          bundle_id: "1",
          success: true
        },
        {
          prescription_id: prescriptionId2,
          bundle_id: "2",
          success: false
        },
        {
          prescription_id: prescriptionId3,
          bundle_id: "3",
          success: false
        }
      ]
    }
  })

  const container = await renderPage()

  expect(screen.getByText(prescriptionId1)).toBeTruthy()
  expect(screen.getByText(prescriptionId2)).toBeTruthy()
  expect(screen.getByText(prescriptionId3)).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

async function renderPage() {
  const {container} = renderWithContext(<SendPage token={token}/>, context)
  await waitFor(() => screen.getByText(/Send Result/))
  return container
}

function clone<T>(value: T): T {
  return JSON.parse(JSON.stringify(value))
}
