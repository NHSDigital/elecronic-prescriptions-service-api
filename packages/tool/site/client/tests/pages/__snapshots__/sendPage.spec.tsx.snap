// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays confirmation page if multiple prescriptions are sent successfully 1`] = `""`;

exports[`Displays confirmation page if single prescription is sent successfully 1`] = `""`;

exports[`Exception report button not shown if there are pending prescriptions 1`] = `""`;

exports[`Exception report button shown if there are failed prescriptions 1`] = `""`;
